lib/collection/src/operations/conversions.rs

```diff
@@ -132,6 +138,165 @@ impl From<Record> for api::grpc::qdrant::RetrievedPoint {
     }
 }
 
+impl TryFrom<api::grpc::qdrant::RetrievedPoint> for Record {
+    type Error = Status;
+
+    fn try_from(retrieved_point: api::grpc::qdrant::RetrievedPoint) -> Result<Self, Self::Error> {
+        Ok(Self {
+            id: retrieved_point.id.unwrap().try_into()?,
+            payload: Some(proto_to_payloads(retrieved_point.payload)?),
+            vector: Some(retrieved_point.vector),
+        })
+    }
+}
+
+impl TryFrom<api::grpc::qdrant::GetCollectionInfoResponse> for CollectionInfo {
+    type Error = Status;
+
+    fn try_from(
+        collection_info_response: api::grpc::qdrant::GetCollectionInfoResponse,
+    ) -> Result<Self, Self::Error> {
+        match collection_info_response.result {
+            None => Err(Status::invalid_argument("Malformed CollectionInfo type")),
+            Some(collection_info_response) => {
+                Ok(Self {
+                    status: match collection_info_response.status {
+                        1 => CollectionStatus::Green,
+                        2 => CollectionStatus::Yellow,
+                        3 => CollectionStatus::Red,
+                        _ => {
+                            return Err(Status::invalid_argument("Malformed CollectionStatus type"))
+                        }
+                    },
```

Comment:
Maybe implement `TryFrom<CollectionStatus> for u32` where CollectionStatus is defined?